### PR TITLE
feat(my-followings): followings page is ready for test

### PR DESCRIPTION
**Pull request by DiLiUA**
- created my followings page component and routes for it;
 - updated  my events page component
 - update Event model;
 - update show info component;
 - update search service;

### CODE REVIEW

File: src/shared/show-info/show-info.component.html [hunk 1]
```diff
@@ -13,30 +13,72 @@
...
+
+      <div class="show-creator" (click)="gotoAnotherPage(false)">by <span>{{show?.creator}}</span></div>
     </div>
 
-    <div class="show-creator">by <span>{{show?.creator}}</span></div>
-    <div class="show-decription">{{sliceDescription(242)}} <a routerlink="/show-page" class="read-more" >Read more »</a></div>
+    <ng-template #atristInfo>
+      <div class="info-top">
+        <div class="left-part">
+          <span class="show-name" (click)="gotoAnotherPage(false)">{{user?.username}}</span>
```
Comment (Ugmaxie) [inline src/shared/show-info/show-info.component.html]: (click)="gotoAnotherPage(false)" ?

File: src/shared/show-info/show-info.component.html [hunk 2]
```diff
@@ -13,30 +13,72 @@
...
+          <div class="genres-container">
+            <span class="genres-title">Genres:</span>
+            <span *ngFor="let genre of user?.genres; last as isLast">
+              {{isLast ? genre : genre + ', '}}
+            </span>
+          </div>
+        </div>
+
+        <div class="right-part">
+          <app-follow *ngIf="!isEvent" [followedUser]="user._id"></app-follow>
```
Comment (Ugmaxie) [inline src/shared/show-info/show-info.component.html]: if user ID is a string - remove dual binding

File: src/shared/show-info/show-info.component.html [hunk 3]
```diff
@@ -13,30 +13,72 @@
...
+          <app-statistics [statistics]="statistics" [id]="user._id" [isEvent]="isEvent"></app-statistics>
```
Comment (Ugmaxie) [inline src/shared/show-info/show-info.component.html]: the same

File: src/shared/show-info/show-info.component.ts [hunk 4]
```diff
@@ -1,24 +1,56 @@
...
   templateUrl: './show-info.component.html',
   styleUrls: ['../../my-events/styles.scss', './show-info.component.scss']
 })
 
-export class ShowInfoComponent implements OnInit {
-  @Input() show: LaunchEvent;
+export class ShowInfoComponent {
+  @Input() set info(data: ShowInfo) {
+    if (!data) {
+      return;
```
Comment (Ugmaxie) [inline src/shared/show-info/show-info.component.ts:23]: return undefined

File: src/shared/show-info/show-info.component.ts [hunk 5]
```diff
@@ -1,24 +1,56 @@
...
+
+    if (!this.isEvent) {
+      this.user = data.user;
+      this.statistics = {
+        likes: this.user.statistics.likes.liked,
+        followers: this.user.statistics.followers,
+        viewers: this.user.statistics.viewers,
+        shows: this.user.shows.owned
+      };
+      return;
```
Comment (Ugmaxie) [inline src/shared/show-info/show-info.component.ts:38]: return undefined

**Review by Ugmaxie — CHANGES_REQUESTED** [review]

File: src/shared/show-info/show-info.component.ts (hunk 5 above)
Comment (DiLiUA) [reply src/shared/show-info/show-info.component.ts:38]: if I do it I have TS err `TS2408: Setters cannot return a value.`

File: src/shared/show-info/show-info.component.ts (hunk 4 above)
Comment (DiLiUA) [reply src/shared/show-info/show-info.component.ts:23]: if I do it I have TS err `TS2408: Setters cannot return a value.`

Comment (Ugmaxie) [reply src/shared/show-info/show-info.component.ts:23]: got it

File: src/event-launch/event-launch.model.ts [hunk 6]
```diff
@@ -71,9 +71,9 @@ export class Show {
       fundedPercentage: 0,
     };
     this.statistics = eventData ? eventData.statistics : {
-      likes: [],
-      viewers: [],
-      followers: []
+      likes: 0,
```
Comment (Ugmaxie) [inline src/event-launch/event-launch.model.ts]: seems it's better to set it as a separate const.

File: src/my-events/my-events.component.html [hunk 7]
```diff
@@ -1,18 +1,13 @@
-<div class="custom-container" style="margin-bottom: 20px">
-  <div class="page-header">
-    <ul class="flex-container flex-center">
-      <li><a href="#" routerLink="/my-events" routerLinkActive="active-link">My events</a></li>
-      <li><a href="#">Following</a></li>
-    </ul>
+<div class="custom-container">
+  <app-my-pages-header [isCreateBtn]="true"></app-my-pages-header>
```
Comment (Ugmaxie) [inline src/my-events/my-events.component.html]: avoid dual-binding
set true in html not good idea

**Review by Ugmaxie — CHANGES_REQUESTED** [review]

**Review by Ugmaxie — APPROVED** [review]

**Pull request merged**